### PR TITLE
RELATED: RAIL-4158 fixes in KPI filter config

### PR DESCRIPTION
**Pull request by no23reason**
* make attribute filter changes optimistic
* make date filter changes optimistic
* improve naming
* fix some bugs

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: libs/sdk-ui-dashboard/src/presentation/widget/common/configuration/useAttributeFilterConfigurationHandling.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    useDashboardCommandProcessing,
+} from "../../../../model";
+import { safeSerializeObjRef } from "../../../../_staging/metadata/safeSerializeObjRef";
+
+export function useAttributeFilterConfigurationHandling(
+    widget: IWidget,
+    displayFormRef: ObjRef,
+    onAppliedChanged: (applied: boolean) => void,
+) {
+    const [status, setStatus] = useState<"ok" | "error" | "loading">("ok");
```
Comment (ivanmjartan) [inline libs/sdk-ui-dashboard/src/presentation/widget/common/configuration/useAttributeFilterConfigurationHandling.ts:19]: Pls change OK status to success :-)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8880/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10725/ : SUCCESS in 15m 10s

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2770

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2541/ : SUCCESS in 1m 18s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2840/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1669/ : SUCCESS in 21m 04s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10728/ : SUCCESS in 11m 51s

**Pull request merged**